libs/scim_proto/src/user.rs

```diff
@@ -0,0 +1,196 @@
+use crate::ScimEntry;
+use base64urlsafedata::Base64UrlSafeData;
+use std::fmt;
+use url::Url;
+use uuid::Uuid;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+struct Name {
+    // The full name including all middle names and titles
+    formatted: Option<String>,
+    family_name: Option<String>,
+    given_name: Option<String>,
+    middle_name: Option<String>,
+    honorific_prefix: Option<String>,
+    honorific_suffix: Option<String>,
+}
+
+/*
+// https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.5
+//
+// https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry
+// Same as locale?
+#[derive(Serialize, Deserialize, Debug, Clone)]
+enum Language {
+    en,
+}
+*/
+
+// https://datatracker.ietf.org/doc/html/rfc5646
+#[allow(non_camel_case_types)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
+enum Locale {
+    en,
+    #[serde(rename = "en-AU")]
+    en_AU,
+    #[serde(rename = "en-US")]
+    en_US,
+    de,
+    #[serde(rename = "en-DE")]
+    de_DE,
+}
+
+impl fmt::Display for Locale {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Locale::en => write!(f, "en"),
+            Locale::en_AU => write!(f, "en-AU"),
+            Locale::en_US => write!(f, "en-US"),
+            Locale::de => write!(f, "de"),
+            Locale::de_DE => write!(f, "de-DE"),
+        }
+    }
+}
+
+#[allow(non_camel_case_types)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
+enum Timezone {
```

Comment:
We should probably use [chrono-tz](https://docs.rs/chrono-tz/latest/chrono_tz/enum.Tz.html) instead of reimplementing it ourselves

Reply:
Oh I didn't know that was a thing lol. I might do that in another PR since the current tz behaviour isn't really relied on anywhere.

Reply:
Yeah, was like "HEY WHAT HOW IS THIS NOT AN IMPORT"

Reply:
Because I'd never found something that could do it :) 

Also FML the fact we need chrono AND time to do ... time. WHY.